### PR TITLE
Fixed: Lexer and added more tests

### PR DESCRIPTION
**Pull request by noob77777**
# Fixed: Some issues with Lexer
Improved lexer.
## Description
Improved Lexer, added namespace jucc::lexer, added more tests.
Improved codecov for lexer.cpp towards 100%
Updated grammar Parser.

## Remaining tasks
- [ ] Review Required

### CODE REVIEW

File: src/lexer/lexer.cpp [hunk 1]
```diff
@@ -9,6 +9,11 @@ int Lexer::GetToken(std::ifstream &is) {
     is.get(last_char);
   }
 
+  // return TOK_EOF if end of file is reached.
+  if (is.eof()) {
+    return TOK_EOF;
+  }
+
```
Comment (TheSYNcoder) [inline src/lexer/lexer.cpp:16]: This was nice, I was thinking of implementing this.
Thanks.

File: test/lexer/comments.txt [hunk 2]
```diff
@@ -5,8 +5,8 @@
...
 // Another comments
 
-int main(){
+int main() {
     // Just another comment 
     // Basically a hello world program, Peace!
     cout << "Hello";
-}
\ No newline at end of file
+}
```
Comment (TheSYNcoder) [inline test/lexer/comments.txt]: I think the earlier version was better as it portrayed that an un-sanitized input file.

**Comment by TheSYNcoder** [comment]
Please don't give branch names as patches.

File: test/lexer/comments.txt (hunk 2 above)
Comment (noob77777) [reply test/lexer/comments.txt]: ok will fix this

**Comment by noob77777** [comment]
> Please don't give branch names as patches.

:angry:

**Comment by TheSYNcoder** [comment]
This looks good!

**Review by TheSYNcoder — APPROVED** [review]

**Comment by TheSYNcoder** [comment]
Please delete the branch after merging. Please 🥰

**Comment by noob77777** [comment]
> 
> 
> Please delete the branch after merging. Please 🥰

No

**Pull request merged**